code/LINDA/LINDA_fire.dm

```diff
@@ -26,13 +26,24 @@
 	if((exposed_temperature > PLASMA_MINIMUM_BURN_TEMPERATURE) && air_contents.toxins > 0.5)
 		igniting = 1
 
+	var/liquid
+	for(var/obj/effect/decal/cleanable/liquid_fuel/L in src)
+		igniting = 1
+		liquid = 1
+		if(exposed_temperature < 700)
+			exposed_temperature = 700
+		if(exposed_volume < 500)
+			exposed_volume = 500
+
 	if(igniting)
-		if(air_contents.oxygen < 0.5 || air_contents.toxins < 0.5)
+		if(air_contents.oxygen < 0.5)
```

Comment:
Actually in this case if you pay attention it ALREADY checks for toxins a bit earlier. Without the toxins it doesn't igniting. Call this an optimization ;)